pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -902,6 +903,16 @@ func defaultGCP(m *Machine, config *admissionConfig) (bool, []string, utilerrors
 
 	providerSpec.Disks = defaultGCPDisks(providerSpec.Disks, config.clusterID)
 
+	if len(providerSpec.GuestAccelerators) != 0 {
+		if providerSpec.GuestAccelerators[0].AcceleratorCount == 0 {
+			providerSpec.GuestAccelerators[0].AcceleratorCount = defaultGCPAcceleratorCount
```

Comment:
Well the idea is that 0 represents unset `AcceleratorCount` value. Which we want to want to default to 1 since we dont expect anybody to define a gpu type with its count as zero, there would be no point. This is for a case when user inputs the type and no count because they might assume it defualts to 1.  

A comment can be added for sure.